### PR TITLE
Feature/cxspa 1115

### PR DESCRIPTION
**Pull request by JedidiahAwuku**
Presentational component for a cart.

### CODE REVIEW

File: feature-libs/pickup-in-store/components/container/pdp-pickup-options-container/pdp-pickup-options-container.spec.ts [hunk 1]
```diff
@@ -266,4 +266,22 @@ describe('PickupOptionsComponent', () => {
...
+            getPickupOption: () => of(null),
+          },
+        })
+        .compileComponents();
+      stubServiceAndCreateComponent();
+    });
+    it('should onInit with no intended location', () => {
+      component.ngOnInit();
+    });
+  });
```
Comment (SVendittelli) [inline feature-libs/pickup-in-store/components/container/pdp-pickup-options-container/pdp-pickup-options-container.spec.ts]: Add expectations to this or remove it.

File: feature-libs/pickup-in-store/components/container/pickup-delivery-info-container/pickup-delivery-info-container.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  beforeEach(() => {
+    fixture = TestBed.createComponent(PickupDeliveryInfoContainerComponent);
+    component = fixture.componentInstance;
+    activeCartService = TestBed.inject(ActiveCartFacade);
+    pickupLocationsSearchService = TestBed.inject(PickupLocationsSearchFacade);
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
```
Comment (SVendittelli) [inline feature-libs/pickup-in-store/components/container/pickup-delivery-info-container/pickup-delivery-info-container.component.spec.ts]:
```suggestion
    expect(component).toBeDefined();
```

File: feature-libs/pickup-in-store/components/container/pickup-delivery-info-container/pickup-delivery-info-container.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    spyOn(pickupLocationsSearchService, 'loadStoreDetails').and.callThrough();
+    spyOn(pickupLocationsSearchService, 'getStoreDetails').and.callThrough();
+    component.ngOnInit();
+    expect(activeCartService.getActive).toHaveBeenCalled();
+    expect(pickupLocationsSearchService.loadStoreDetails).toHaveBeenCalledWith(
+      'London School'
+    );
+    expect(pickupLocationsSearchService.getStoreDetails).toHaveBeenCalledWith(
+      'London School'
+    );
```
Comment (SVendittelli) [inline feature-libs/pickup-in-store/components/container/pickup-delivery-info-container/pickup-delivery-info-container.component.spec.ts:81]: Add an assertion to what `storesDetailsData` is set to after running these.

File: feature-libs/pickup-in-store/components/container/pickup-delivery-info-container/pickup-delivery-info-container.component.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  IntendedPickupLocationFacade,
+  PickupLocationsSearchFacade,
+} from '@spartacus/pickup-in-store/root';
+import { combineLatest, Observable } from 'rxjs';
+import { filter, map, mergeMap, tap } from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-pickup-delivery-info-container',
+  templateUrl: './pickup-delivery-info-container.component.html',
+  styleUrls: ['./pickup-delivery-info-container.component.scss'],
```
Comment (SVendittelli) [inline feature-libs/pickup-in-store/components/container/pickup-delivery-info-container/pickup-delivery-info-container.component.ts]: We don't do styling at the component level, remove this line, and it's file

File: feature-libs/pickup-in-store/components/presentational/pickup-delivery-info/pickup-delivery-info.component.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { Component, Input } from '@angular/core';
+import { PointOfService } from '@spartacus/core';
+
+@Component({
+  selector: 'cx-pickup-delivery-info',
+  templateUrl: './pickup-delivery-info.component.html',
+  styleUrls: ['./pickup-delivery-info.component.scss'],
```
Comment (SVendittelli) [inline feature-libs/pickup-in-store/components/presentational/pickup-delivery-info/pickup-delivery-info.component.ts]: We don't use styles per component, remove this reference.

File: feature-libs/pickup-in-store/components/presentational/pickup-delivery-info/pickup-delivery-info.component.scss [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+      display: flex;
+      .store-hours-day {
+        width: 50%;
+      }
+      .store-hours-time {
+        width: 50%;
+      }
+    }
+  }
+}
```
Comment (SVendittelli) [inline feature-libs/pickup-in-store/components/presentational/pickup-delivery-info/pickup-delivery-info.component.scss]: Move this to `feature-libs/pickup-in-store/styles` and update the styles index

File: feature-libs/pickup-in-store/components/presentational/pickup-delivery-info/pickup-delivery-info.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(PickupDeliveryInfoComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
```
Comment (SVendittelli) [inline feature-libs/pickup-in-store/components/presentational/pickup-delivery-info/pickup-delivery-info.component.spec.ts]:
```suggestion
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { I18nTestingModule } from '@spartacus/core';
import { PickupDeliveryInfoComponent } from './pickup-delivery-info.component';

describe('PickupDeliveryInfoComponent', () => {
  let component: PickupDeliveryInfoComponent;
  let fixture: ComponentFixture<PickupDeliveryInfoComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [PickupDeliveryInfoComponent],
      imports: [I18nTestingModule],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(PickupDeliveryInfoComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeDefined();
```

File: feature-libs/pickup-in-store/components/presentational/pickup-delivery-info/pickup-delivery-info.module.ts [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+import { ReactiveFormsModule } from '@angular/forms';
+import { I18nModule } from '@spartacus/core';
+import { StoreModule } from '../store/store.module';
+import { PickupDeliveryInfoComponent } from './pickup-delivery-info.component';
+
+@NgModule({
+  imports: [CommonModule, I18nModule, ReactiveFormsModule, StoreModule],
```
Comment (SVendittelli) [inline feature-libs/pickup-in-store/components/presentational/pickup-delivery-info/pickup-delivery-info.module.ts]:
```suggestion
  imports: [CommonModule, I18nModule, StoreModule],
```

File: feature-libs/pickup-in-store/components/presentational/store/store-address/store-address.component.spec.ts [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+import { IconTestingModule } from '@spartacus/storefront';
+import { StoreAddressComponent } from './store-address.component';
+
+describe('StoreAddressComponent', () => {
+  let component: StoreAddressComponent;
+  let fixture: ComponentFixture<StoreAddressComponent>;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [I18nTestingModule, IconTestingModule],
```
Comment (SVendittelli) [inline feature-libs/pickup-in-store/components/presentational/store/store-address/store-address.component.spec.ts]:
```suggestion
      imports: [I18nTestingModule],
```

File: feature-libs/pickup-in-store/core/store/effects/pickup-location.spec.ts [hunk 10]
```diff
@@ -0,0 +1,70 @@
+import { HttpClientTestingModule } from '@angular/common/http/testing';
```
Comment (SVendittelli) [inline feature-libs/pickup-in-store/core/store/effects/pickup-location.spec.ts]: File name should be `feature-libs/pickup-in-store/core/store/effects/pickup-location.effect.spec.ts`

**Review by SVendittelli — CHANGES_REQUESTED** [review]

File: feature-libs/pickup-in-store/components/container/pdp-pickup-options-container/pdp-pickup-options-container.spec.ts (hunk 1 above)
Comment (JedidiahAwuku) [reply feature-libs/pickup-in-store/components/container/pdp-pickup-options-container/pdp-pickup-options-container.spec.ts]: Removed

File: feature-libs/pickup-in-store/components/container/pickup-delivery-info-container/pickup-delivery-info-container.component.spec.ts (hunk 2 above)
Comment (JedidiahAwuku) [reply feature-libs/pickup-in-store/components/container/pickup-delivery-info-container/pickup-delivery-info-container.component.spec.ts]: Done

File: feature-libs/pickup-in-store/components/container/pickup-delivery-info-container/pickup-delivery-info-container.component.spec.ts (hunk 3 above)
Comment (JedidiahAwuku) [reply feature-libs/pickup-in-store/components/container/pickup-delivery-info-container/pickup-delivery-info-container.component.spec.ts:81]: Done

File: feature-libs/pickup-in-store/components/presentational/pickup-delivery-info/pickup-delivery-info.component.ts (hunk 5 above)
Comment (JedidiahAwuku) [reply feature-libs/pickup-in-store/components/presentational/pickup-delivery-info/pickup-delivery-info.component.ts]: Done

File: feature-libs/pickup-in-store/components/presentational/pickup-delivery-info/pickup-delivery-info.component.scss (hunk 6 above)
Comment (JedidiahAwuku) [reply feature-libs/pickup-in-store/components/presentational/pickup-delivery-info/pickup-delivery-info.component.scss]: Done

File: feature-libs/pickup-in-store/components/presentational/pickup-delivery-info/pickup-delivery-info.component.spec.ts (hunk 7 above)
Comment (JedidiahAwuku) [reply feature-libs/pickup-in-store/components/presentational/pickup-delivery-info/pickup-delivery-info.component.spec.ts]: Done

File: feature-libs/pickup-in-store/components/presentational/pickup-delivery-info/pickup-delivery-info.module.ts (hunk 8 above)
Comment (JedidiahAwuku) [reply feature-libs/pickup-in-store/components/presentational/pickup-delivery-info/pickup-delivery-info.module.ts]: Done

File: feature-libs/pickup-in-store/components/presentational/store/store-address/store-address.component.spec.ts (hunk 9 above)
Comment (JedidiahAwuku) [reply feature-libs/pickup-in-store/components/presentational/store/store-address/store-address.component.spec.ts]: Done

File: feature-libs/pickup-in-store/core/store/effects/pickup-location.spec.ts (hunk 10 above)
Comment (JedidiahAwuku) [reply feature-libs/pickup-in-store/core/store/effects/pickup-location.spec.ts]: Done

**Review by SVendittelli — APPROVED** [review]

**Pull request merged**